core/lib/resources/presence/presence_manager.js

```diff
@@ -39,13 +40,30 @@ PresenceManager.prototype.setup = function() {
   });
 
   // save so you removeListener on destroy
-  this.sentryListener = function(sentry) {
+  this.sentryListener = function (sentry) {
+    if (manager.destroying) return;
+
     var clientIds = store.clientsForSentry(sentry);
 
     var chain = function (clientIds) {
+      var immediateObject;
       var clientId = clientIds.pop();
 
-      if (!clientId) return;
+      // Pick up any new users/clients that have subscribed to this resource
+      if (!clientId && !manager.destroying) {
+        clientIds = store.clientsForSentry(sentry);
+        clientId = clientIds.pop();
+      }
+
+      if (!clientId || manager.destroying) {
+        if (immediateObject) {
+          clearImmediate(immediateObject);
+        }
+
+        logging.info('#presence - #sentry down chain exit: clientId:',
```

Comment:
Since sentry down is rare enough, and more rare still when manager.destroying === true, I will let this as an *info* logging statement.  As it is, another more verbose statement in chain() is already logging at the info level.